rasterio/plot.py

```diff
@@ -327,9 +334,25 @@ def show_hist(
         plt.show()
 
 
+def hist_strech(arr, clip_percent):
+    """
+    Histogram streching for better image visualization.
+   Parameters
+    ----------
+    arr : array-like in the image form of (rows, columns, bands)
+        image to reshape
+    clip_percent: the minimum values (cumulative percentage) of the histogram,
+                  and the maximum value (cumulative percentage) is set to 100% - clip_percent
+                  default clip_percent is 2.
+    """
+    arr_hist = np.nanpercentile(np.array(arr), (clip_percent, 100 - clip_percent))
+    arr = (arr - arr_hist[0]) / (arr_hist[1] - arr_hist[0])
+    arr = np.clip(arr, 0, 1)
```

Comment:
Is it necessary to clip the array? Is it not ensured by the math on the line above?

Reply:
For rgb image show, it is not necessary, while for the single-band image show, clip the array is necessary. the test sample can be found at: https://github.com/xinluo2018/rasterio/blob/main/test.ipynb